zhmc_prometheus_exporter/zhmc_prometheus_exporter.py

```diff
@@ -1730,8 +1751,151 @@ def collect(self):
         for family_obj in family_objects.values():
             yield family_obj
 
+        end_dt = datetime.now()
+        duration = (end_dt - start_dt).total_seconds()
+        if self.last_export_dt:
+            self.export_interval = \
+                (end_dt - self.last_export_dt).total_seconds()
+        self.last_export_dt = end_dt
+        interval_str = "{:.1f} sec".format(self.export_interval) if \
+            self.export_interval else "None"
         logprint(logging.INFO, None,
-                 "Done collecting metrics")
+                 "Done collecting metrics after {:.1f} sec (export interval: "
+                 "{})".format(duration, interval_str))
+
+    def run_fetch_thread(self, session):
+        """
+        Function that runs as the property fetch thread.
+
+        Requires to be run against HMC 2.16.0, because it uses:
```

Comment:
With zhmcclient 1.14.0, that is no longer correct - it is supported for all HMC versions. It is just faster with HMC 2.16.0.